### DISCUSSION TITLE
Create boilerplate files if not found

### DISCUSSION THREAD

**Discussion by elibroftw** [wailsapp/wails#1259] [General]
**Is your feature request related to a problem? Please describe.**

I'll detail my experiences so you know where I'm coming from. I started a Tauri project from scratch earl January. I used the create-react-app for the frontend and made a video on the entire experience. Then last month, I started using Mantine. Today, I started developing the actual application. At this point, their build was working fine.

Today, I encountered an issue with Tauri's notification, so I asked and was told to update Tauri because a new version had come out.
I updated but I wanted to add a custom animation for when buttons are pressed `SELECT:active:hover { transform: translateY(4px) }`.
This caused noticeable lag between navigation clicks in Tauri but not in Firefox nor in Edge. So I said I'll build the app to test it out like that. My computer crashed completely so when I turned it on to build, Tauri said to delete the build files. So I did that and the build succeeded but the output binary was unstyled. I then figured out how to test a built react app locally and it looked fine. So it was a Tauri issue. Then I said, okay, this might be an issue deeper than Tauri, possibly with wry or Webview2 itself.

Then I tried to use Python to test out the react app. Big yikes. `eel` is 2 years out of date, `cefpython` doesn't work on Python 3.8, and neither does `pywebview`. So then I said, okay, I had starred wails, let me check it out.

Aside from wails init not working how git init does ("Executing git init creates a . git subdirectory in the current working directory, which contains all of the necessary Git metadata for the new repository"), there's no easy way to transfer an existing web app to use wails. 

So I transferred the source code files, found a better way than editing `yarn start` to work for Wails, and ran `wails dev`. my gitignore file ignores everything in the build directory because that's where React's output goes. I'm not sure how Vue and etc, do it, but that's how React does it. So I'm not going to rely on copy pasting an appicon.png because if I clone the repo to my laptop, the code won't run!

**Describe the solution you'd like**

Just copy the init files on runtime if they were not found.

**Describe alternatives you've considered**
Will try to use webview or lorca lol

**Additional context**

**Comment by elibroftw** [comment]
Also, https://wails.io/docs/gettingstarted/firstproject, should mention community templates.

**Comment by elibroftw** [comment]
Is there a way to tell wails to use a URL for source rather `wails dev`?

**Comment by elibroftw** [comment]
ERROR:
main.go:13:12: pattern frontend/dist: no matching files found

There should really be an option to not require a dist directory if running with custom dev watcher.

**Comment by leaanthony** [comment]
I'm not sure that's this project is going to give you what you want, but anyhoo:
1. There's some tight coupling between project layout and the Wails CLI. I'm not sure what just generating boilerplate files into a react project would really mean. Care to elaborate?
2. Will update the firstproject page with your suggestion. The website does have an "edit this page" link if you find anything else you think should be added.
3. Regarding the URL for source, you can find plenty on this in the discussions board and issue tracker. Bottom line: it's in demand but nobody wants to own this and make it happen, even with my help.
4. I'm not sure how you're going to fix that considering it's a compile time Go requirement. I mean, we don't use it but we can't guard it either. Open to suggestions on this.

**Comment by elibroftw** [comment]
1. I mean if wails can't detect an app icon, just create a default app icon.
3. What do you mean by own it?
4. What's a compile time go requirement? (I don't know what you were responding to)

**Comment by leaanthony** [comment]
> 1. I mean if wails can't detect an app icon, just create a default app icon.

Yeah, thanks for highlighting that. It should have been doing it but there was a regression that I've fixed on master. Just be aware that `build/windows/icon.ico` will need deleting if you update your `appicon.png`.

> 2. What do you mean by own it?

Someone keen to provide feedback on how it should work, keen to test PRs I/we create and see it through to completion. 

> 3. What's a compile time go requirement? (I don't know what you were responding to)

Yeah sorry, it's late here. I was responding to this:

> ERROR: main.go:13:12: pattern frontend/dist: no matching files found
> 
> There should really be an option to not require a dist directory if running with custom dev watcher.
